src/winscard_svc.c

```diff
@@ -886,47 +886,52 @@ static LONG MSGRemoveContext(SCARDCONTEXT hContext, SCONTEXT * threadContext)
 		hCard = *(int32_t *)ptr;
 
 		/*
-		 * Unlock the sharing
+		 * Unlock the sharing. If the reader or handle already disappeared, skip
+		 * the disconnection part and just delete the orphan handle.
 		 */
 		rv = RFReaderInfoById(hCard, &rContext);
-		if (rv != SCARD_S_SUCCESS)
+		if (rv != SCARD_S_SUCCESS && rv != SCARD_E_READER_UNAVAILABLE &&
+			 rv != SCARD_E_INVALID_VALUE)
 		{
 			(void)pthread_mutex_unlock(&threadContext->cardsList_lock);
 			return rv;
 		}
 
-		if (0 == rContext->hLockId)
+		if (rContext)
```

Comment:
The diff is weird here, but essentially I just wrapped the whole block of code below into "if (rContext)".
Maybe there's a more elegant way to change the code.